### PR TITLE
accession checks

### PR DESCRIPTION
**Pull request by davidrichardson**
accessions should normally only be provided if USI already knows about
them. If accessions are provided, identity details should match with
the known records.

### CODE REVIEW

File: src/main/java/uk/ac/ebi/subs/api/services/UserAuthoritiesService.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import java.util.stream.Stream;
+
+@Component
+public class UserAuthoritiesService {
+
+    public Stream<String> userAuthoritiesStream() {
+        final Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
+        if (authentication == null) {
+            return Stream.of();
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/api/services/UserAuthoritiesService.java]: Is `Stream.empty()` not better?

File: src/main/java/uk/ac/ebi/subs/api/validators/CoreSubmittableValidationHelper.java [hunk 2]
```diff
@@ -1,21 +1,18 @@
...
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.data.domain.PageRequest;
 import org.springframework.stereotype.Component;
 import org.springframework.validation.Errors;
-import org.springframework.validation.ValidationUtils;
 import uk.ac.ebi.subs.api.services.OperationControlService;
-import uk.ac.ebi.subs.data.status.ProcessingStatusEnum;
-import uk.ac.ebi.subs.data.submittable.Submittable;
+import uk.ac.ebi.subs.api.services.UserAuthoritiesService;
+import uk.ac.ebi.subs.api.services.UserTeamService;
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/api/validators/CoreSubmittableValidationHelper.java]: I think we don't use this import

File: src/main/java/uk/ac/ebi/subs/api/validators/CoreSubmittableValidationHelper.java [hunk 3]
```diff
@@ -62,26 +64,70 @@ public void validate(StoredSubmittable target, SubmittableRepository repository,
...
 
     private void ensureChecklistIsForSameDataTypeAsSubmittable(StoredSubmittable target, Errors errors) {
         DataType dataType = target.getDataType();
         Checklist checklist = target.getChecklist();
 
         if (dataType != null && checklist != null
-                && !dataType.getId().equals( checklist.getDataTypeId())){
-            SubsApiErrors.invalid.addError(errors,"checklist");
+                && !dataType.getId().equals(checklist.getDataTypeId())) {
+            SubsApiErrors.invalid.addError(errors, "checklist");
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/api/validators/CoreSubmittableValidationHelper.java]: Is this error not related to `dataType` instead of `checklist`?

File: src/main/java/uk/ac/ebi/subs/api/services/UserAuthoritiesService.java (hunk 1 above)
Comment (davidrichardson) [reply src/main/java/uk/ac/ebi/subs/api/services/UserAuthoritiesService.java]: agreed

File: src/main/java/uk/ac/ebi/subs/api/validators/CoreSubmittableValidationHelper.java (hunk 2 above)
Comment (davidrichardson) [reply src/main/java/uk/ac/ebi/subs/api/validators/CoreSubmittableValidationHelper.java]: good catch 👍 🐟

File: src/main/java/uk/ac/ebi/subs/api/validators/CoreSubmittableValidationHelper.java (hunk 3 above)
Comment (davidrichardson) [reply src/main/java/uk/ac/ebi/subs/api/validators/CoreSubmittableValidationHelper.java]: the check is for inconsistency between dataType and checklist, I think it's reasonable to report it for the checklist as this is typically part of the request body, where the datatype is typically given in the URL

Comment (ke4) [reply src/main/java/uk/ac/ebi/subs/api/validators/CoreSubmittableValidationHelper.java]: Agreed
